provider/adapters/src/fallback/mod.rs

```diff
@@ -309,72 +336,64 @@ impl LocaleFallbacker {
         }
     }
 
-    /// Creates the intermediate [`LocaleFallbackerWithConfig`] with configuration options.
+    /// Creates an iterator based on a [`Locale`].
+    ///
+    /// When first initialized, the locale is normalized according to the fallback algorithm.
+    ///
+    /// For performance reasons, internally an [`icu_provider::DataLocale`] is used.
+    ///
+    /// [`Locale`]: icu_locid::Locale
+    #[doc(hidden)] // for now
+    pub fn fallback_for(
+        &self,
+        config: LocaleFallbackConfig,
+        locale: impl Into<DataLocale>,
```

Comment:
Nit (optional): Still not the hugest fan of `impl Into` arguments; just let people write `.into()` at the call site for the same effect and no need to import extra types